src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.particle.ParticleManager;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumBlockRenderType;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import net.minecraftforge.client.event.DrawBlockHighlightEvent;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.property.ExtendedBlockState;
+import net.minecraftforge.common.property.IExtendedBlockState;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.fmp.ForgeMultipart;
+import net.minecraftforge.fmp.client.multipart.AdvancedParticleManager;
+import net.minecraftforge.fmp.client.multipart.ICustomHighlightPart;
+import net.minecraftforge.fmp.client.multipart.MultipartStateMapper;
+import net.minecraftforge.fmp.microblock.IMicroblockContainerTile;
+import net.minecraftforge.fmp.microblock.MicroblockContainer;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.Multipart;
+import net.minecraftforge.fmp.multipart.MultipartContainer;
+import net.minecraftforge.fmp.multipart.MultipartRegistry;
+import net.minecraftforge.fmp.multipart.PartState;
+import net.minecraftforge.fmp.raytrace.RayTraceUtils;
+
+/**
+ * An {@link Block} that allows your block to act as a microblock container.<br/>
+ * Extend this class if you want your block to be able to have microblocks, but not any other kinds of multiparts.
+ * Otherwise, extend {@link Multipart} or make a custom implementation of {@link IMultipart}.<br/>
+ * All the overriden methods have a "default" counterpart that allows you to handle interactions with your block if the
+ * player isn't interacting with a multipart.<br/>
+ * Extend {@link TileCoverable} or implement {@link IMicroblockContainerTile} and return it in
+ * {@link BlockCoverable#createNewTileEntity(World, int)} if you want a custom tile entity for your block.
+ */
```

Comment:
The purpose of BlockCoverable is to allow people to keep using a normal block but have the ability to place microblocks in the same space as that block. It's a "light" solution to adding multipart support that, while keeping your Block and TE like they were, allows you to have the extra microblock support most mods want.
The only way I can think of that would be able to get around this (without removing it completely) would be some kind of "layered logic" system where the default behavior (vanilla's) would be the bottommost layer and developers would be able to build logic on top of that which falls back to the next layer if the current one doesn't support it. This way, there would be the "vanilla" layer, then the "microblock container" layer and finally all the blocks' logic layers. I'm not sure how reliable and fast a system like that would be, but I think this is a fairly simple and good solution for now.